### PR TITLE
EEITT-1182 New auth lib

### PR DESCRIPTION
**Pull request by tenochdlc**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/auth/models/Retrievals.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  affinityGroup: Option[AffinityGroup],
+  internalId: Option[String],
+  externalId: Option[String],
+  userDetails: UserDetails,
+  credentialStrength: Option[String],
+  agentCode: Option[String]
+)
+
+object Retrievals {
+  def getTaxIdValue(maybeEnrolment: Option[String], taxIdName: String)(implicit retrievals: Retrievals) = {
```
Comment (pawelpanasewicz) [inline app/uk/gov/hmrc/gform/auth/models/Retrievals.scala]: Could we pass retrievals explicitly? We tend to reduce of `implicit` usage in gform.

File: app/uk/gov/hmrc/gform/auth/models/Retrievals.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    val maybeEnrolmentIdentifier = maybeEnrolment match {
+      case Some(enrolment) => retrievals.enrolments.getEnrolment(enrolment)
+        .fold[Option[EnrolmentIdentifier]](None)(_.getIdentifier(taxIdName))
```
Comment (pawelpanasewicz) [inline app/uk/gov/hmrc/gform/auth/models/Retrievals.scala]: how about replacing 
```scala
.fold[Option[EnrolmentIdentifier]](None)(_.getIdentifier(taxIdName))
```
with 
```scala
.map(_.getIdentifier(taxIdName))
```

File: app/uk/gov/hmrc/gform/auth/EeittAuthorisationDelegate.scala [hunk 3]
```diff
@@ -17,27 +17,31 @@
...
-    } yield isAuthed(isOk)
-
-  private def isAuthed(isOk: Verification): AuthResult = {
-    if (!isOk.isAllowed)
-      NeedsAuthenticated
-    else
-      Authenticated
+class EeittAuthorisationDelegate(eeittConnector: EeittConnector, configModule: ConfigModule) {
+
+  def legacyAuth(formTemplate: FormTemplate, userDetails: UserDetails)(implicit hc: HeaderCarrier, ex: ExecutionContext, request: Request[AnyContent]): Future[Result] = {
```
Comment (lutnos) [inline app/uk/gov/hmrc/gform/auth/EeittAuthorisationDelegate.scala]: Agreed

File: app/uk/gov/hmrc/gform/auth/models/Retrievals.scala (hunk 1 above)
Comment (lutnos) [inline app/uk/gov/hmrc/gform/auth/models/Retrievals.scala]: I'll have a look at what is involved to do that

**Pull request merged**